### PR TITLE
feat(TV UI): Navigation rail redesign

### PR DESCRIPTION
**Pull request by KingLucius**
- New icons (Home & Settings).
- Filled icons on focus & selecting (Home, Library & Settings).
- Account selection at bottom of the NavigrationRail (TV Only).

This PR is part of large change (notification center on TV) but decided to push this to make the review easier so you will find some comments (notification badge on updates) and a header on NavigationRail not used.

image

### CODE REVIEW

**Review by fire-light42 — CHANGES_REQUESTED** [review]
`accounts` may be `arrayOf<Account>()` aka emptyList in case the user has never used the account feature. This makes the account profile pic empty when it should use the default account profile pic.

File: app/src/main/java/com/lagradost/cloudstream3/MainActivity.kt
Comment (fire-light42) [inline app/src/main/java/com/lagradost/cloudstream3/MainActivity.kt:1683]: No, why are you doing this here when you have the observe. Remove the homeViewModel.currentAccount shit and do everything here or use observe properly. I would recommend doing it here as accounts are not tied to the home view.

Comment (KingLucius) [reply app/src/main/java/com/lagradost/cloudstream3/MainActivity.kt:1683]: I am using the observe to update after the account management dialog and here to update on start up

Comment (KingLucius) [reply app/src/main/java/com/lagradost/cloudstream3/MainActivity.kt:1683]: You mean I should observe here to be able to update across all instead of home?

Comment (fire-light42) [reply app/src/main/java/com/lagradost/cloudstream3/MainActivity.kt:1683]: Observe updates instantly if you have set the value beforehand. Right now I am just worried that this code can lead to more bugs like this, as the account you observe does not include the default account.

Comment (fire-light42) [reply app/src/main/java/com/lagradost/cloudstream3/MainActivity.kt:1683]: So stick to one approach and not both, either make observe the only thing that changes the UI or only change UI with the callback event.

**Review by fire-light42 — CHANGES_REQUESTED** [review]
I want `So stick to one approach and not both, either make observe the only thing that changes the UI or only change UI with the callback event.` to be fixed before merge.

**Comment by KingLucius** [comment]
> I want `So stick to one approach and not both, either make observe the only thing that changes the UI or only change UI with the callback event.` to be fixed before merge.

I partly finished this, testing it before pushing

**Review by fire-light42 — APPROVED** [review]
Code looks good, will merge after testing 👍

**Pull request merged**
